test/jdk/sun/security/krb5/Krb5NameEquals.java

```diff
@@ -22,57 +22,145 @@
+                }
+            } else {
+                result.shouldHaveExitValue(0);
+            }
+        } else {
+
+            final var testCommand = new String[]{"-Djava.security.krb5.realm=R",
+                    "-Djava.security.krb5.kdc=127.0.0.1",
+                    Krb5NameEquals.Krb5NameEqualsTest.class.getName()
+            };
+
+            final var result = ProcessTools.executeTestJava(testCommand);
+            result.shouldHaveExitValue(0);
         }
-        MECH = temp;
     }
 
-    public static void main(String[] argv) throws Exception {
-        GSSManager mgr = GSSManager.getInstance();
-
-        boolean result = true;
-        // Create GSSName and check their equals(), hashCode() impl
-        GSSName name1 = mgr.createName(NAME_STR1,
-            GSSName.NT_HOSTBASED_SERVICE, MECH);
-        GSSName name2 = mgr.createName(NAME_STR2,
-            GSSName.NT_HOSTBASED_SERVICE, MECH);
-        GSSName name3 = mgr.createName(NAME_STR1,
-            GSSName.NT_HOSTBASED_SERVICE, MECH);
-
-        if (!name1.equals(name1) || !name1.equals(name3) ||
-            !name1.equals((Object) name1) ||
-            !name1.equals((Object) name3)) {
-            System.out.println("Error: should be the same name");
-            result = false;
-        } else if (name1.hashCode() != name3.hashCode()) {
-            System.out.println("Error: should have same hash");
-            result = false;
+    static class Krb5NameEqualsTest {
+
+        private static String NAME_STR1 = "service@localhost";
+        private static String NAME_STR2 = "service2@localhost";
+        private static final Oid MECH;
+
+        static {
+            Oid temp = null;
+            try {
+                temp = new Oid("1.2.840.113554.1.2.2"); // KRB5
+            } catch (Exception e) {
+                // should never happen
+                throw new RuntimeException("Exception initialising Oid", e);
+            }
+            MECH = temp;
         }
 
-        if (name1.equals(name2) || name1.equals((Object) name2)) {
-            System.out.println("Error: should be different names");
-            result = false;
+        public static void main(String[] argv) throws Exception {
+            GSSManager mgr = GSSManager.getInstance();
+
+            boolean result = true;
+            // Create GSSName and check their equals(), hashCode() impl
+            GSSName name1 = mgr.createName(NAME_STR1,
+                    GSSName.NT_HOSTBASED_SERVICE, MECH);
+            GSSName name2 = mgr.createName(NAME_STR2,
+                    GSSName.NT_HOSTBASED_SERVICE, MECH);
+            GSSName name3 = mgr.createName(NAME_STR1,
+                    GSSName.NT_HOSTBASED_SERVICE, MECH);
+
+            if (!name1.equals(name1) || !name1.equals(name3) ||
+                !name1.equals((Object) name1) ||
+                !name1.equals((Object) name3)) {
+                System.out.println("Error: should be the same name");
```

Comment:
You can just throw exceptions and the exit code of the Java process with be 1 and the stack trace will be in the log.